src/components/TextInput.tsx

```diff
@@ -14,46 +13,36 @@ type TextInputProps = {
   value: string;
   type: string;
   isError: boolean;
-  backgroundColor?: string;
   className?: string;
   onChange: (value: string) => void;
 };
 
 export const TextInput = (props: TextInputProps) => {
-  const [active, setActive] = useState(false);
-
   return (
-    <div className={cn("relative h-11", props.className)}>
+    <div className={cn("relative h-11 ", props.className)}>
       <h2
         className={twMerge(
           cn(
-            "absolute left-3 transition-all pointer-events-none z-10",
-            props.backgroundColor,
+            "absolute left-3 transition-all pointer-events-none z-10 bg-background px-2",
           ),
-          active || props.value != ""
-            ? "top-[-0.75rem] px-2 text-sm  text-blue-500"
-            : "top-1.5 cursor-text text-gray-500",
-          props.isError &&
-            "absolute left-3 top-[-0.75rem] px-2 text-sm  text-red-500",
+          props.value != ""
+            ? "top-[-0.75rem] text-sm text-blue-500"
+            : "top-1.5 cursor-text text-tertiary/70 dark:text-white/70",
```

Comment:
with the approach detailed in the video, you shouldn't need to use the value of the prop to change style, and it can be 100% done with css selectors